proposals/NNNN-unlock-existentials-with-fixed-associated-types.md

```diff
@@ -0,0 +1,237 @@
+
+```swift
+let myUser: User
+// ❌ `User` is not
+// usable as a Type
+```
+> **_NOTE:_** One may point out that `myUser` could be bound to `some User`, which would have a similar effects. However, [opaque result types](https://github.com/apple/swift-evolution/blob/master/proposals/0244-opaque-result-types.md) are not Existentials. That is, the former allows for hiding type information from the user while the compiler internally retains the underlying type. On the contrary, the latter properly erases type information allowing storage of different types - that conform to a given protocol. What that means, is that should `myUser` be bound to `some User` then initialization would have to be performed at the declaration-site and mutation would be prohibited.
+
+This is a great inconvenience with not so elegant workarounds:
+
+```swift
+protocol AnyUser { ... } 
+// No `Identifiable` conformance
+
+let myUser: AnyUser
+// ✅
+
+
+protocol User: AnyUser, Identifiable 
+    where ID == String { 
+    
+    var id: String {
+        username
+    }
+}
+
+let myOtherUser: User 
+// ❌ Still an error as expected
+```
+
+These workarounds, besides producing boilerplate and confusing code, also produce confusing API designs leaving clients to wonder which of the two protocols should be used. In other cases, API authors may decide that adding `Identifiable` isn’t worth the added complexity and leave the `User` protocol with no such conformance whatsoever. 
+
+As a result, a straightforward conformance to simple - yet quite useful - protocols such as `Identifiable` and others becomes a hassle to implement.
+
+
+## Proposed solution
+
+We propose to simply allow User-like protocols to be used as Types. Thus, making more natural code possible - which fits a goal of Swift of building ["expressive and elegant APIs"](https://forums.swift.org/t/on-the-road-to-swift-6/32862). Furthermore, library authors could start adding useful protocol conformances to their protocols - a task that's currently prohibitively complex.
+
+
+## Detailed design
+
+### Which Protocols would NOT be Usable as Types 
+
+Now, a protocol is considered unusable as a Type when it:
+
+1. Includes at least one associated type or `Self` requirement - which may have been inherited - and
+2. if at least one inherited associated type requirement - if there is any - is _not_ specified.
+
+
+### Protocols that would be Usable as Types
+
+1. Fixed Associated Types
+```swift
+protocol AB { 
+    associatedtype A
+    associatedtype B
+}
+
+protocol FixedAB: AB
+    where A == String, B == String {}
+    
+let foo: FixedAB 
+// ✅ 
+```
+> **_NOTE:_** Note that both `A` and `B` were fixed in order for `FixedAB` to be usable as a Type.
+
+2. Fixed Associated Types in Composition
+```swift
+protocol C { 
+    associatedtype C
+}
+
+protocol FixedC: C
+    where C == String {}
+
+typealias FixedABC = FixedAB & FixedC
+
+let foo: FixedABC 
+// ✅ 
+```
+
+3. Fixed Associated Types in Composition with a Class
```

Comment:
Rather than trying to compose potentially confusing names for each example, I think we should arrange the examples according to the kind of existential type (protocol/protocol composition/subclass existential)